hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -63,6 +63,10 @@ object InferPType {
         val ptype = PType.canonical(t, false)
         ptype.deepInnerRequired(false)
       }
+      case Die(_, t) => {
```

Comment:
moved closer to If to make it apparent that they have same behavior.

Reply:
this can actually be top-level required too.